File: src/main/java/net/hectus/neobb/turn/default_game/throwable/TEnderPearl.java
package net.hectus.neobb.turn.default_game.throwable;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.CounterFilter;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.attributes.function.CounterFunction;
import org.bukkit.Location;
import org.bukkit.entity.Projectile;
import org.bukkit.potion.PotionEffectType;

import java.util.List;

public class TEnderPearl extends ThrowableTurn implements CounterFunction, NeutralClazz {
    public TEnderPearl(NeoPlayer player) { super(player); }
    public TEnderPearl(Projectile data, Location impact, NeoPlayer player) { super(data, impact, player); }

    @Override
    public int cost() {
        return 3;
    }

    @Override
    public void apply() {
        player.player.removePotionEffect(PotionEffectType.LEVITATION);
        new Buff.ExtraTurn().apply(player);
    }

    @Override
    public List<CounterFilter> counters() {
        return List.of(CounterFilter.of(t -> t instanceof TSplashLevitationPotion, "levitation"), CounterFilter.of(turn -> turn.getClass().getSimpleName().endsWith("Wall"), "walls"));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/throwable/TEnderPearl.kt
package net.hectus.neobb.modes.turn.default_game.throwable

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.modes.turn.default_game.CounterFilter
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.CounterFunction
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.entity.Projectile
import org.bukkit.potion.PotionEffectType

class TEnderPearl(data: Projectile?, cord: Cord?, player: NeoPlayer?) : ThrowableTurn(data, cord, player), CounterFunction, NeutralClazz {
    override val cost: Int = 3

    override fun apply() {
        player!!.player.removePotionEffect(PotionEffectType.LEVITATION)
        ExtraTurn().apply(player)
    }

    override fun counters(): List<CounterFilter> {
        return listOf(
            CounterFilter.of("levitation") { it is TSplashLevitationPotion },
            CounterFilter.of("walls") { turn -> turn::class.simpleName!!.endsWith("Wall") }
        )
    }
}
